bambi/interpret/effects.py

```diff
@@ -1,890 +1,857 @@
-# pylint: disable=ungrouped-imports
-from dataclasses import dataclass, field
-import itertools
+from typing import Any, Callable, Mapping, Optional
 
 import arviz as az
 import numpy as np
 import pandas as pd
-from pandas.api.types import is_categorical_dtype, is_string_dtype
-import xarray as xr
+from arviz import InferenceData
+from pandas import DataFrame, Series
+from pandas.api.types import (
+    is_float_dtype,
+    is_integer_dtype,
+    is_numeric_dtype,
+)
+from xarray import DataArray
 
-from bambi.models import Model
-from bambi.interpret.create_data import create_differences_data, create_predictions_data
+from bambi import Model
 from bambi.interpret.utils import (
-    average_over,
-    ConditionalInfo,
-    enforce_dtypes,
+    aggregate,
+    create_plot_config,
+    get_model_covariates,
+    get_response_and_target,
     identity,
-    merge,
-    VariableInfo,
 )
-from bambi.utils import get_aliased_name
-
 
-SUPPORTED_SLOPES = ("dydx", "eyex")
-SUPPORTED_COMPARISONS = {
-    "diff": lambda x, y: x - y,
-    "ratio": lambda x, y: x / y,
-}
+from .helpers import compare, create_inference_data
+from .plots import plot
+from .types import (
+    Conditional,
+    ConditionalParam,
+    Contrast,
+    ContrastParam,
+    Values,
+    Variable,
+)
 
 
-@dataclass
-class ResponseInfo:
-    """Stores metadata about the response variable for indexing data in az.InferenceData,
-    computing uncertainty intervals, and creating the summary dataframe in
-    'PredictiveDifferences'
+def validate_values(
+    values: Values, var_name: str, target_dtype: np.dtype | None = None
+) -> Series:
+    """Validate and convert user provided values to a Pandas Series.
```

Comment:
Are we converting to a pandas series because that is what seaborn prefers/expects? Have you encountered any serious drawback because of having to work with Series?

Reply:
Perfect